src/rules/deprecationRule.ts

```diff
@@ -210,23 +207,23 @@ function getDeprecationFromDeclarations(declarations?: ts.Declaration[]): string
     if (declarations === undefined) {
         return undefined;
     }
-    let declaration: ts.Node;
-    for (declaration of declarations) {
-        if (isBindingElement(declaration)) {
-            declaration = getDeclarationOfBindingElement(declaration);
-        }
-        if (isVariableDeclaration(declaration)) {
-            declaration = declaration.parent;
-        }
-        if (isVariableDeclarationList(declaration)) {
-            declaration = declaration.parent;
-        }
-        const result = getDeprecationFromDeclaration(declaration);
-        if (result !== undefined) {
-            return result;
-        }
+
+    const deprecations = declarations
+        .map(declaration => getDeprecationFromDeclaration(getRootDeclaration(declaration)))
+        .filter(deprecation => deprecation !== undefined);
+    const allDeclarationsDeprecated = deprecations.length === declarations.length;
+    return allDeclarationsDeprecated ? deprecations.join(", ") : undefined;
```

Comment:
When there are multiple deprecated overloads, I concatenate all their descriptions, not sure if that's the best approach.

Reply:
Let's just go with the first deprecation description. These can be repeated and/or long and/or multiline; it'd be confusing to get a whole bunch of test spam from a few joined descriptions.